rust/rope/src/spans.rs

```diff
@@ -209,3 +224,53 @@ impl<'a, T: Clone + Default> Iterator for SpanIter<'a, T> {
         None
     }
 }
+
+impl<'a, T: Clone + Default> Iterator for SpanBackwardsIter<'a, T> {
+    type Item = (Interval, &'a T);
+
+    fn next(&mut self) -> Option<(Interval, &'a T)> {
+        if self.done {
+            return None;
+        }
+        if let Some((leaf, start_pos)) = self.cursor.get_leaf() {
+            if leaf.spans.is_empty() { return None; }
+            let leaf_start = self.cursor.pos() - start_pos;
+            let span = &leaf.spans[leaf.spans.len() - self.ix - 1];
+            self.ix += 1;
+            if self.ix == leaf.spans.len() {
+                if self.cursor.prev_leaf().is_none() {
+                    self.done = true;
```

Comment:
This is currently a workaround, because if there are leafs, `self.cursor.get_leaf()` seems to never return `None`. This is maybe a bug in `cursor.prev_leaf()`, or I am doing something wrong.